packages/react-native-performance/android/src/main/java/com/oblador/performance/PerformanceModule.java

```diff
@@ -121,33 +143,77 @@ private void setupMarkerListener() {
         );
     }
 
-    private void safelyEmitMark(String name, long startTime) {
+    private void safelyEmitMark(PerformanceEntry entry) {
         if (eventsBuffered) {
-            markBuffer.put(name, startTime);
+            addMark(entry);
         } else {
-            emitMark(name, startTime);
+            emitMark(entry);
         }
     }
 
+    private static void addMark(PerformanceEntry entry) {
+        markBuffer.add(entry);
+    }
+
     private void emitBufferedMarks() {
-        for (Map.Entry<String, Long> entry : markBuffer.entrySet()) {
-            emitMark(entry.getKey(), entry.getValue());
+        didEmit = true;
+        for (PerformanceEntry entry : markBuffer) {
+            emitMark(entry);
         }
+        emitNativeBufferedMarks();
     }
 
-    private void emitMark(String name,
-                          long startTime) {
-        emit("mark", name, startTime);
+    private void emitNativeBufferedMarks() {
+        for (PerformanceEntry entry : RNPerformance.getInstance().getEntries()) {
+            emitMark(entry);
+        }
+    }
+
+    private void emitMark(PerformanceEntry entry) {
+        if (entry instanceof PerformanceMark) {
+            emit("mark", (PerformanceMark) entry);
+        } else if (entry instanceof PerformanceMetric) {
+            emit("metric", (PerformanceMetric) entry);
+        }
     }
 
-    private void emit(String eventName,
-                      String name,
-                      long startTime) {
+    private void emit(String eventName, PerformanceMetric metric) {
```

Comment:
Is `eventName` the right parameter name? Maybe we don't even need this parameter, since for emitting metrics the name will always be `metric`, and same for marks.